cmd/flowserver/server.go

```diff
@@ -269,11 +269,19 @@ func (s *FlowServer) handleMigrate(w http.ResponseWriter, r *http.Request) (inte
 		return nil, fmt.Errorf("missing flows element")
 	}
 
-	flows, err := definition.ReadLegacyFlows(migrate.Flows)
+	legacyFlows, err := legacy.ReadLegacyFlows(migrate.Flows)
```

Comment:
previously flow migration only fully happened when re-marshalling back to JSON. Now you get a `legacy.Flow` with a `Migrate` method. I think this is simpler and makes it easier to apply the same validation rules to migrated flows.